src/Compilers/Extension/SetGlobalProjectCollectionForCPS.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.Build;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Roslyn.Compilers.Extension
+{
+    [ExportBuildGlobalPropertiesProvider]
+    [AppliesTo("(" + ProjectCapabilities.CSharp + " | " + ProjectCapabilities.VB + ")" + " & " + ProjectCapabilities.LanguageService)]
+    public class SetGlobalGlobalPropertiesForCPS : StaticGlobalPropertiesProviderBase
+    {
+        [ImportingConstructor]
+        public SetGlobalGlobalPropertiesForCPS(IProjectService projectService)
+            : base(projectService.Services)
+        {
+        }
+
+        public override Task<IImmutableDictionary<string, string>> GetGlobalPropertiesAsync(CancellationToken cancellationToken)
+        {
+            return CompilerPackage.RoslynHive != null
```

Comment:
Is there a timing issue here where the CompilerExtensions package would be loaded after this? I think it's okay given the CompilerExtension is an auto-load package based on Solution Explorer hence this should be okay in the general case.